fix(semantic): `IdentifierReference` within `TSPropertySignature` cannot reference type-only import binding

Dunqing opened the pull request:
close: #5435

The behavior of `IdentifierReference` in `TSPropertySignature` is the same as in `TSTypeQuery`, both allow only reference value bindings and type-only import bindings.

I still use `ReferenceFlags::TSTypeQuery` here because I want to avoid producing many changes unrelated to the bug in this PR. I will refactor it in the follow-up PR soon

Dunqing commented:
* **#5441** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5441?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented:
Should we add a test case to semantic's integration tests? maybe in `symbols.rs`?

Dunqing commented:
> Should we add a test case to semantic's integration tests? maybe in `symbols.rs`?

I added a fixture for testing, and as you can see from the snapshot the results are as expected，So I think that's enough.

Dunqing committed:
fix(semantic): `IdentifierReference` within `TSPropertySignature` cannot reference type-only import binding (#5441)

close: #5435

The behavior of `IdentifierReference` in `TSPropertySignature` is the same as in `TSTypeQuery`, both allow only reference value bindings and type-only import bindings.

I still use `ReferenceFlags::TSTypeQuery` here because I want to avoid producing many changes unrelated to the bug in this PR. I will refactor it in the follow-up PR soon
